### DISCUSSION TITLE
Periodic BC on cubit generated, 3D mesh

### DISCUSSION THREAD

**Discussion by bhollra** [idaholab/moose#17267] [Q&A General] [answered]
Hello,
I am working on a steady state 3D solid heat conduction problem. I would like to implement a periodic boundary condition in my model, but adding the following into the `[BCs]` section of my input file causes the residuals to stagnate well before an appropriate convergence tolerance.

> `[./Periodic]`
>`	[./x]`
>`		variable = T_solid`
>`		primary = left`
>`		secondary = right`
>`		translation = '1 0 0'`
>`	[../]`
>`	[./y]`
>`		variable = T_solid`
>`		primary = front`
>`		secondary = back`
>`		translation = '0 1 0'`
>`	[../]`
>`[../]`

To demonstrate this, I have created a simpler version of my case where the issue is recreated. In a case where I remove the periodic BC, the solution has no trouble solving, as shown below:
![non_periodic_result](https://user-images.githubusercontent.com/49657641/110573441-e2445e00-8120-11eb-988f-1423ebb9e933.png)

Any insight into any potential mistakes in my implementation or a more appropriate method for implementing a periodic BC for this case would be much appreciated.  
Brent

Attached are the input file and mesh I am using. 
[periodic_bc.zip](https://github.com/idaholab/moose/files/6112982/periodic_bc.zip)

**Comment by bhollra** [comment]
It seems there was an error running the original files I uploaded with MOOSE (I was using SAM to run them with no issues). Attached are updated files, tested with MOOSE only, which reproduce the issue described above.

[periodic_bc_1.zip](https://github.com/idaholab/moose/files/6118561/periodic_bc_1.zip)

**Comment by GiudGiud** [comment]
Tagging @permcody @andrsd who should know a lot about periodic BCs

**Comment by andrsd** [comment, marked as answer]
Your problem is in your PETSc options. When I comment them out, I get this convergence:
```
 0 Nonlinear |R| = 5.744566e+00
      0 Linear |R| = 5.744566e+00
      1 Linear |R| = 3.609398e-01
      2 Linear |R| = 8.977029e-02
      3 Linear |R| = 1.969694e-02
      4 Linear |R| = 1.959308e-02
      5 Linear |R| = 1.350552e-02
      6 Linear |R| = 1.279056e-03
      7 Linear |R| = 7.000715e-04
      8 Linear |R| = 3.249532e-04
      9 Linear |R| = 8.143903e-05
     10 Linear |R| = 2.449361e-05
 1 Nonlinear |R| = 2.449361e-05
      0 Linear |R| = 2.449361e-05
      1 Linear |R| = 2.279705e-05
      2 Linear |R| = 2.255615e-05
      3 Linear |R| = 2.128266e-05
      4 Linear |R| = 1.389275e-05
      5 Linear |R| = 4.483888e-06
      6 Linear |R| = 9.006490e-07
      7 Linear |R| = 3.828971e-07
      8 Linear |R| = 1.659223e-07
      9 Linear |R| = 7.817783e-08
     10 Linear |R| = 4.299557e-08
     11 Linear |R| = 2.047900e-08
     12 Linear |R| = 8.901540e-09
     13 Linear |R| = 4.517054e-09
     14 Linear |R| = 1.658610e-09
     15 Linear |R| = 7.478177e-10
     16 Linear |R| = 3.790151e-10
     17 Linear |R| = 9.637592e-11
 2 Nonlinear |R| = 9.637587e-11
 Solve Converged!
```
And this result:
![image](https://user-images.githubusercontent.com/85632/111656509-352cae00-87d0-11eb-9fac-ccd6b7884baf.png)

**Reply by bhollra** [reply]
Thanks for your response. Removing the PETSc options did the trick for me!

**Comment by mangerij** [comment]
Very nice mesh... what sort of CUBIT options did you use to do this?

I always have difficulty getting periodic nodes when using the tetrahedrons.

**Reply by bhollra** [reply]
I output the blocks using the "hex8" element type, but it looks like the actual output is a combination of hexahedral and wedge element types, so I don't have any experience using tetrahedrons with a periodic boundary. 

Here is the .jou file I used if you're interested. 
[test_mesh.txt](https://github.com/idaholab/moose/files/6223832/test_mesh.txt)
